### ISSUE TITLE
New rapid `set_text` code seems broken in 3.32.0

### ISSUE THREAD

**Issue by voidfiles** [teamcapybara/capybara#2332]
##  Meta
Capybara Version: 3.32.0
Driver Information (and browser if relevant):
chromedriver: 80.0.3987.106

## Expected Behavior
Given the HTML 
```
<input type="text" id="shipping-address" maxlength="50">
```

I would expect the follow test to work

```
fill_in 'shipping-address', with: ('a' * 55)
expect(find('#shipping-address').value.length).to eq 50
```

## Actual Behavior
Given the HTML 
```
<input type="text" id="shipping-address" maxlength="50">
```

Oddly, the follow test works, but shouldn't.

```
fill_in 'shipping-address', with: ('a' * 55)
expect(find('#shipping-address').value.length).to eq 52
```

## Workarounds

Given the HTML 
```
<input type="text" id="shipping-address" maxlength="50">
```

If I write the test like this, turning off rapid text set, it works.

```
fill_in 'shipping-address', with: ('a' * 55), fill_options: {rapid: false}
expect(find('#shipping-address').value.length).to eq 50
```

After looking at the changes between 3.31.0 and 3.32.0 I expect something is going [on here](https://github.com/teamcapybara/capybara/blame/e62d8484d7ccf22e2143cf8e0b5fe0f4cafa3f6f/lib/capybara/selenium/node.rb#L282-L285)

At a bare minimum I think the slicing logic is wrong, if I pop open a IRB and run the following, you can see the the "10" is in repeated in two slices.

```
2.6.3 :001 > a = %w(1 2 3 4 5 6 7 8 9 10 11 12)
 => ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"]
2.6.3 :002 > a[0..3]
 => ["1", "2", "3", "4"]
2.6.3 :003 > a[-3..-1]
 => ["10", "11", "12"]
2.6.3 :004 > a[4..-3]
 => ["5", "6", "7", "8", "9", "10"]
2.6.3 :005 > a[-3..-1]
 => ["10", "11", "12"]
```

**Comment by twalpole** [comment]
@voidfiles The reason you're seeing `10` repeated is because you've done [4..-3] whereas the code does [4...-3] (inclusive vs exclusive ranges), but yes there does seem to be an issue with length

**Comment by voidfiles** [comment]
Ah, thanks for pointing that out.

**Comment by twalpole** [comment]
Please try the rapid_test branch and see if that solves your issue.

**Comment by voidfiles** [comment]
I just verified the rapid_test branch. It fixes the issue.

**Comment by twalpole** [comment]
Fixed by PR #2333
